MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```
@@ -18,14 +18,22 @@ final class FeedViewModel {
 		Localized.Feed.title
 	}
 
+	var errorMessage: String {
+		Localized.Feed.loadError
+	}
+
 	var onLoadingStateChange: Observer<Bool>?
+	var onLoadingErrorOccur: Observer<Bool>?
```

`onLoadingErrorOccur: Observer<Bool>?` could be `onErrorStateChange: Observer<String?>?`, so the ViewModel decides which message to show and when to hide errors (instead of the view controller). It's easier to manage presentation state transitions when all changes are centralized.

For example, in the `loadFeed()` method, the ViewModel can call  `onErrorStateChange(.none)` to hide the error before loading the feed.